OpenRA.Game/Server/Server.cs

```diff
@@ -341,10 +343,18 @@ void ValidateClient(Connection newConn, string data)
 					return;
 				}
 
+				var ipAddress = ((IPEndPoint)newConn.Socket.RemoteEndPoint).Address;
+
+				string location = null;
+				if (!string.IsNullOrEmpty(Settings.GeoIPDatabase))
+					location = Session.IsLocalNetworkAddress(ipAddress) ? "Local Network" : GeoIP.LookupCountry(ipAddress.ToString());
```

Comment:
If I read correctly (not tested) each locally hosted server would require the `GeoIPDatabase` just to display "Local Network". Imho it would be much better to not require a downloaded database for local games.

Reply:
No, it will silently fail. You don't require the GeoIP database at all:

https://github.com/OpenRA/OpenRA/blob/df61b59746978a7187e0c5f39f7d584e249805a7/OpenRA.Game/Network/GeoIP.cs#L59-L68

Reply:
Won't `Settings.GeoIPDatabase` be empty (or null), so we never enter this branch?

Reply:
My logic here was that it would be weird to show "Local Network" for local players, but nothing for non-local players. Better to never show anything for anyone...? (this seems logical to me, but i'm not convinced it is best)

Reply:
> Won't `Settings.GeoIPDatabase` be empty (or null), so we never enter this branch?

The error message somehow advertises the feature.

Reply:
Not sure what you mean. The point was that the if check prevents us ever calling `LookupCountry` in the first place if we have no database.
My initial concern was that this in turn means that almost every locally created server will not display "Local Network" for all the local clients (and there are no public clients here). However, my good-morning-awakening-thoughts changed my opinion to preferring to drop the entire "Local Network" thing since if I read the code right (despite my morning tiredness), you'd have to download the database and set the path in your `settings.yaml` just to display "Local Network" for lan games - silly!
That made me think in which cases we'd have players displaying "Local Network":
- Most of the time in local lan games where this is unnecessary information, since the game is local by definition.
- Perhaps if someone was to host a public game from their client. Displaying "Local Network" to other players here is wrong/misleading since the hosting person is not on their local network.
- Someone hosting a public dedicated server joining in from their local network. Again displaying "Local Network" would be wrong/misleading for the same reason.

In the last case this might even completely give the clients identity (IP) away, since you know the public IP address of the server, and that the client is in the same local network, so for example on the same machine or behind the same NAT.

If I'm not completely missing something here, it sounds logical to me to just drop this feature and reduce the code complexity.

Reply:
Ok. Removed "Local Network" and "Unknown Location" labels. All cases now simply omit the label on the client side.